### PR TITLE
Improve node vsphere node deletion

### PR DESCRIPTION
**Pull request by daxmc99**
Problem: Vsphere node deletion entirely relies on the local directory to
ensure uniqueness during node deletion. Pratically, this means the name
is the only unique identifier used. This can be an issue when the
local dir is being modified out of the bounds of docker machine.

Solution: The vsphere driver now uses the ssh key fingerprints of the
keys to ensure the node being deleted is the intended node.

Addresses: https://github.com/rancher/rancher/issues/20971

### CODE REVIEW

File: drivers/vmwarevsphere/vsphere.go [hunk 1]
```diff
@@ -965,6 +970,51 @@ func (d *Driver) Remove() error {
...
+	opman := guest.NewOperationsManager(c.Client, vm.Reference())
+
+	fileman, err := opman.FileManager(ctx)
+	if err != nil {
+		return err
+	}
+
+	src := d.ResolveStorePath("auth_keys")
+	auth := AuthFlag{}
+	auth.auth.Username = B2DUser
```
Comment (luthermonson) [inline drivers/vmwarevsphere/vsphere.go]: should use defaultSSHUser and defaultSSHPass, my updates will make everything standardize to this.

Comment (daxmc99) [reply drivers/vmwarevsphere/vsphere.go]: Updated, thanks

File: drivers/vmwarevsphere/vsphere.go [hunk 2]
```diff
@@ -8,6 +8,7 @@ import (
 	"archive/tar"
 	"encoding/base64"
 	"fmt"
+	ssh2 "golang.org/x/crypto/ssh"
```
Comment (dramich) [inline drivers/vmwarevsphere/vsphere.go]: Formatting

File: drivers/vmwarevsphere/vsphere.go [hunk 3]
```diff
@@ -1106,6 +1150,15 @@ func (d *Driver) fetchVM(ctx context.Context, c *govmomi.Client, vmname string)
 	return vm, nil
 }
 
+func getPublicKeyFingerprint(pubKeyBytes []byte) (string,error) {
```
Comment (dramich) [inline drivers/vmwarevsphere/vsphere.go]: Formatting

File: drivers/vmwarevsphere/vsphere.go [hunk 4]
```diff
@@ -965,6 +970,51 @@ func (d *Driver) Remove() error {
...
+	remoteKeyBytes, err := ioutil.ReadFile(src)
+	if err != nil {
+		return err
+	}
+	remoteKeyprint, err := getPublicKeyFingerprint(remoteKeyBytes)
+	if err != nil {
+		return err
+	}
+	if remoteKeyprint != localKeyprint{
+		return fmt.Errorf("local key fingerprint is different than remote key fingerprint")
```
Comment (dramich) [inline drivers/vmwarevsphere/vsphere.go]: Should not be `Errorf`

**Comment by dramich** [comment]
lgtm

**Pull request merged**